src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -35,8 +35,29 @@ pub struct WorkUnit {
   pub span_id: String,
 }
 
-pub trait WorkUnitStore {
-  fn should_record_zipkin_spans(&self) -> bool;
+pub trait WorkUnitStore: Send + Sync {
   fn get_workunits(&self) -> &Mutex<Vec<WorkUnit>>;
   fn add_workunit(&self, workunit: WorkUnit);
 }
+
+pub struct SafeWorkUnitStore {
```

Comment:
We decided to call it safe because it is a Mutex of a value(Vec<WorkUnit>). So we wanted to say that it is safe to work with it from several threads. 
What could be a better name in this case?

Reply:
Just `WorkUnitStore` - Rust guarantees that you'll always be able to work with things properly across threads (and specifically the `Send` and `Sync` marker traits are used to denote these kinds of safety) :)